extension/entitystore/ec2Info.go

```diff
@@ -0,0 +1,144 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package entitystore
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/aws/amazon-cloudwatch-agent/internal/ec2metadataprovider"
+	"github.com/aws/amazon-cloudwatch-agent/plugins/processors/ec2tagger"
+)
+
+const (
+	// InstanceId character maximum length is 19.
+	// See https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_Instance.html.
+	instanceIdSizeMax = 19
+
+	// AutoScalingGroup character maximum length is 255.
+	// See https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_AutoScalingGroup.html.
+	autoScalingGroupSizeMax = 255
+)
+
+type EC2Info struct {
+	InstanceID       string
+	AccountID        string
+	AutoScalingGroup string
+
+	// region is used while making call to describeTags Ec2 API for AutoScalingGroup
+	Region string
```

Comment:
nit: Is there a reason these are public instead of using getters? Are we modifying them outside of this package?

Reply:
Blocking comment.

Because this helps with mutex locking for variable that are accessed by multiple sources.